src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -317,10 +320,15 @@ otError RadioSpinel<InterfaceType, ProcessContextType>::CheckRadioCapabilities(v
 
     if ((mRadioCaps & kRequiredRadioCaps) != kRequiredRadioCaps)
     {
-        otLogCritPlat("RCP does not support required capabilities: ack-timeout:%s, tx-retries:%s, CSMA-backoff:%s",
-                      (mRadioCaps & OT_RADIO_CAPS_ACK_TIMEOUT) ? "yes" : "no",
-                      (mRadioCaps & OT_RADIO_CAPS_TRANSMIT_RETRIES) ? "yes" : "no",
-                      (mRadioCaps & OT_RADIO_CAPS_CSMA_BACKOFF) ? "yes" : "no");
+        otRadioCaps missingCaps = (mRadioCaps & kRequiredRadioCaps) ^ kRequiredRadioCaps;
+
+        otLogCritPlat("RCP capabilities check failed: ack-timeout:%s, tx-retries:%s, CSMA-backoff:%s, tx-security:%s, "
+                      "tx-timing:%s",
+                      (missingCaps & OT_RADIO_CAPS_ACK_TIMEOUT) ? "failed" : "pass",
```

Comment:
Is this change necessary?

Reply:
If we use the old way. Newly added capabilities are only needed in `OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE`, then we have to add #if defines inside `otLogCritPlat` which will trigger macos-clang check error.

Change it this way will make it more clear that which capability is missing.

Reply:
It makes sense. How about only display missing features:
```cpp
otLogCritPlat("RCP does not support required capabilities:%s%s%s%s",
(missingCaps & OT_RADIO_CAPS_ACK_TIMEOUT) ? " ack-timeout" : "",
(missingCaps & OT_RADIO_CAPS_TRANSMIT_RETRIES) ? " tx-retries" : "",
....
```

Reply:
Sounds good. Code updated.